make && make install error while using cpp-argparse 3.0

```
(xeus-cling) anutosh491@spbhat68:~/QuantStack/xeus-cling/build$ make && make install
[  9%] Building CXX object CMakeFiles/xeus-cling.dir/src/xinterpreter.cpp.o
[ 18%] Building CXX object CMakeFiles/xeus-cling.dir/src/xoptions.cpp.o
[ 27%] Building CXX object CMakeFiles/xeus-cling.dir/src/xmagics/executable.cpp.o
/home/anutosh491/QuantStack/xeus-cling/src/xmagics/executable.cpp: In member function 'xcpp::argparser xcpp::executable::get_options()':
/home/anutosh491/QuantStack/xeus-cling/src/xmagics/executable.cpp:70:16: error: use of deleted function 'xcpp::argparser::argparser(const xcpp::argparser&)'
   70 |         return argpars;
      |                ^~~~~~~
In file included from /home/anutosh491/QuantStack/xeus-cling/src/xmagics/executable.cpp:37:
/home/anutosh491/QuantStack/xeus-cling/include/xeus-cling/xoptions.hpp:21:12: note: 'xcpp::argparser::argparser(const xcpp::argparser&)' is implicitly deleted because the default definition would be ill-formed:
   21 |     struct argparser : public argparse::ArgumentParser
      |            ^~~~~~~~~
/home/anutosh491/QuantStack/xeus-cling/include/xeus-cling/xoptions.hpp:21:12: error: use of deleted function 'argparse::ArgumentParser::ArgumentParser(const argparse::ArgumentParser&)'
In file included from /home/anutosh491/QuantStack/xeus-cling/include/xeus-cling/xoptions.hpp:15,
                 from /home/anutosh491/QuantStack/xeus-cling/src/xmagics/executable.cpp:37:
/home/anutosh491/conda_root/envs/xeus-cling/include/argparse/argparse.hpp:1477:3: note: declared here
 1477 |   ArgumentParser(const ArgumentParser &other) = delete;
      |   ^~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/xeus-cling.dir/build.make:146: CMakeFiles/xeus-cling.dir/src/xmagics/executable.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:166: CMakeFiles/xeus-cling.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```
I think the readme can specify the version to be below `3.0` (currently it says that the version can be somewhere around `2.9`)

on gentoo linux i have a problem compiling with cpp-argparse v2.9 or v3.0

this is with 3.0:

```
/home/ufk/src/xeus-cling/src/xmagics/executable.cpp: In member function 'xcpp::argparser xcpp::executable::get_options()':
/home/ufk/src/xeus-cling/src/xmagics/executable.cpp:70:16: error: use of deleted function 'xcpp::argparser::argparser(const xcpp::argparser&)'
   70 |         return argpars;
      |                ^~~~~~~
In file included from /home/ufk/src/xeus-cling/src/xmagics/executable.cpp:37:
/home/ufk/src/xeus-cling/include/xeus-cling/xoptions.hpp:21:12: note: 'xcpp::argparser::argparser(const xcpp::argparser&)' is implicitly deleted because the default definition would be ill-formed:
   21 |     struct argparser : public argparse::ArgumentParser
      |            ^~~~~~~~~
/home/ufk/src/xeus-cling/include/xeus-cling/xoptions.hpp:21:12: error: use of deleted function 'argparse::ArgumentParser::ArgumentParser(const argparse::ArgumentParser&)'
In file included from /home/ufk/src/xeus-cling/include/xeus-cling/xoptions.hpp:15,
                 from /home/ufk/src/xeus-cling/src/xmagics/executable.cpp:37:
/home/ufk/miniforge3/envs/xeus-cling/include/argparse/argparse.hpp:1477:3: note: declared here
 1477 |   ArgumentParser(const ArgumentParser &other) = delete;
      |   ^~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/xeus-cling.dir/build.make:146: CMakeFiles/xeus-cling.dir/src/xmagics/executable.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:166: CMakeFiles/xeus-cling.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

```

and v2.9:

```
/home/ufk/src/xeus-cling/src/xmagics/executable.cpp: In member function ‘xcpp::argparser xcpp::executable::get_options()’:
/home/ufk/src/xeus-cling/src/xmagics/executable.cpp:70:16: error: use of deleted function ‘xcpp::argparser::argparser(xcpp::argparser&&)’
   70 |         return argpars;
      |                ^~~~~~~
In file included from /home/ufk/src/xeus-cling/src/xmagics/executable.cpp:37:
/home/ufk/src/xeus-cling/include/xeus-cling/xoptions.hpp:21:12: note: ‘xcpp::argparser::argparser(xcpp::argparser&&)’ is implicitly deleted because the default definition would be ill-formed:
   21 |     struct argparser : public argparse::ArgumentParser
      |            ^~~~~~~~~
/home/ufk/src/xeus-cling/include/xeus-cling/xoptions.hpp:21:12: error: use of deleted function ‘argparse::ArgumentParser::ArgumentParser(argparse::ArgumentParser&&)’
In file included from /home/ufk/src/xeus-cling/include/xeus-cling/xoptions.hpp:15:
/home/ufk/miniforge3/envs/xeus-cling/include/argparse/argparse.hpp:1479:3: note: declared here
 1479 |   ArgumentParser(ArgumentParser &&) noexcept = delete;
      |   ^~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/xeus-cling.dir/build.make:146: CMakeFiles/xeus-cling.dir/src/xmagics/executable.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:166: CMakeFiles/xeus-cling.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

@anutosh491 I am also facing the same build error. Did you figure out how to solve this? There's a PR #512 too but not being merged yet. 
I am using Ubuntu 22.04.4. 

```
(xeus-cling) ukiyo@ukiyo-Nitro-AN515-44:~/xeus-cling/build$ cmake && make install
Usage

  cmake [options] <path-to-source>
  cmake [options] <path-to-existing-build>
  cmake [options] -S <path-to-source> -B <path-to-build>

Specify a source directory to (re-)generate a build system for it in the
current working directory.  Specify an existing build directory to
re-generate its build system.

Run 'cmake --help' for more information.

[  9%] Building CXX object CMakeFiles/xeus-cling.dir/src/xmagics/executable.cpp.o
/home/ukiyo/xeus-cling/src/xmagics/executable.cpp: In member function 'xcpp::argparser xcpp::executable::get_options()':
/home/ukiyo/xeus-cling/src/xmagics/executable.cpp:70:16: error: use of deleted function 'xcpp::argparser::argparser(const xcpp::argparser&)'
   70 |         return argpars;
      |                ^~~~~~~
In file included from /home/ukiyo/xeus-cling/src/xmagics/executable.cpp:37:
/home/ukiyo/xeus-cling/include/xeus-cling/xoptions.hpp:21:12: note: 'xcpp::argparser::argparser(const xcpp::argparser&)' is implicitly deleted because the default definition would be ill-formed:
   21 |     struct argparser : public argparse::ArgumentParser
      |            ^~~~~~~~~
/home/ukiyo/xeus-cling/include/xeus-cling/xoptions.hpp:21:12: error: use of deleted function 'argparse::ArgumentParser::ArgumentParser(const argparse::ArgumentParser&)'
In file included from /home/ukiyo/xeus-cling/include/xeus-cling/xoptions.hpp:15,
                 from /home/ukiyo/xeus-cling/src/xmagics/executable.cpp:37:
/home/ukiyo/miniforge3/envs/xeus-cling/include/argparse/argparse.hpp:1477:3: note: declared here
 1477 |   ArgumentParser(const ArgumentParser &other) = delete;
      |   ^~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/xeus-cling.dir/build.make:146: CMakeFiles/xeus-cling.dir/src/xmagics/executable.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:166: CMakeFiles/xeus-cling.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

--- a/README.md
+++ b/README.md
@@ -34,7 +34,7 @@ mamba install xeus-cling -c conda-forge
 You will first need to create a new environment and install the dependencies:
 
 ```bash
-mamba create -n xeus-cling -c conda-forge cmake xeus-zmq cling nlohmann_json cppzmq xtl pugixml doctest cpp-argparse
+mamba create -n xeus-cling -c conda-forge cmake xeus-zmq cling nlohmann_json=3.11.2 cppzmq xtl pugixml doctest cpp-argparse<3.1
 source activate xeus-cling
 ```
 Please refer to [environment-host.yml](https://github.com/jupyter-xeus/xeus-cling/blob/main/environment-host.yml) for packages specific versions to install if applicable.
@@ -107,7 +107,7 @@ A C++ backend for the Jupyter interactive widgets is available in the [`xwidgets
 
 | `xeus-cling` |   `xeus-zmq`    |      `xtl`      |     `cling`   |   `pugixml`   | `cppzmq` | `cpp-argparse`| `nlohmann_json` | `dirent` (windows only) |
 |--------------|-----------------|-----------------|---------------|---------------|----------|---------------|-----------------|-------------------------|
-|  main        |  >=1.1.0,<2.0.0 |  >=0.7.0,<0.8.0 | >=0.9,<0.10   | ~1.8.1        | ~4.3.0   |     ~2.9      | >=3.6.1,<4.0    | >=2.3.2,<3              |
+|  main        |  >=1.1.0,<2.0.0 |  >=0.7.0,<0.8.0 | >=0.9,<0.10   | ~1.8.1        | ~4.3.0   |     <3.1      | ~3.11.2         | >=2.3.2,<3              |
 |  0.15.3      |  >=1.1.0,<2.0.0 |  >=0.7.0,<0.8.0 | >=0.9,<0.10   | ~1.8.1        | ~4.3.0   |     ~2.9      | >=3.6.1,<4.0    | >=2.3.2,<3              |
 |  0.15.2      |  >=1.1.0,<2.0.0 |  >=0.7.0,<0.8.0 | >=0.9,<0.10   | ~1.8.1        | ~4.3.0   |     ~2.9      | >=3.6.1,<4.0    | >=2.3.2,<3              |
 |  0.15.1      |  >=1.0.0,<2.0.0 |  >=0.7.0,<0.8.0 | >=0.9,<0.10   | ~1.8.1        | ~4.3.0   |     ~2.9      | >=3.6.1,<4.0    | >=2.3.2,<3              |

--- a/README.md
+++ b/README.md
@@ -34,7 +34,7 @@ mamba install xeus-cling -c conda-forge
 You will first need to create a new environment and install the dependencies:
 
 ```bash
-mamba create -n xeus-cling -c conda-forge cmake xeus-zmq cling nlohmann_json=3.11.2 cppzmq xtl pugixml doctest cpp-argparse<3.1
+mamba create -n xeus-cling -c conda-forge cmake xeus-zmq cling nlohmann_json=3.11.2 cppzmq xtl pugixml doctest cpp-argparse
 source activate xeus-cling
 ```
 Please refer to [environment-host.yml](https://github.com/jupyter-xeus/xeus-cling/blob/main/environment-host.yml) for packages specific versions to install if applicable.
@@ -107,7 +107,7 @@ A C++ backend for the Jupyter interactive widgets is available in the [`xwidgets
 
 | `xeus-cling` |   `xeus-zmq`    |      `xtl`      |     `cling`   |   `pugixml`   | `cppzmq` | `cpp-argparse`| `nlohmann_json` | `dirent` (windows only) |
 |--------------|-----------------|-----------------|---------------|---------------|----------|---------------|-----------------|-------------------------|
-|  main        |  >=1.1.0,<2.0.0 |  >=0.7.0,<0.8.0 | >=0.9,<0.10   | ~1.8.1        | ~4.3.0   |     <3.1      | ~3.11.2         | >=2.3.2,<3              |
+|  main        |  >=1.1.0,<2.0.0 |  >=0.7.0,<0.8.0 | >=0.9,<0.10   | ~1.8.1        | ~4.3.0   |     ~3.0      | ~3.11.2         | >=2.3.2,<3              |
 |  0.15.3      |  >=1.1.0,<2.0.0 |  >=0.7.0,<0.8.0 | >=0.9,<0.10   | ~1.8.1        | ~4.3.0   |     ~2.9      | >=3.6.1,<4.0    | >=2.3.2,<3              |
 |  0.15.2      |  >=1.1.0,<2.0.0 |  >=0.7.0,<0.8.0 | >=0.9,<0.10   | ~1.8.1        | ~4.3.0   |     ~2.9      | >=3.6.1,<4.0    | >=2.3.2,<3              |
 |  0.15.1      |  >=1.0.0,<2.0.0 |  >=0.7.0,<0.8.0 | >=0.9,<0.10   | ~1.8.1        | ~4.3.0   |     ~2.9      | >=3.6.1,<4.0    | >=2.3.2,<3              |

--- a/environment-host.yml
+++ b/environment-host.yml
@@ -7,11 +7,11 @@ dependencies:
   # Host dependencies
   - xeus-zmq=1.1.0
   - cling=0.9
-  - nlohmann_json
+  - nlohmann_json=3.11.2
   - cppzmq
   - xtl=0.7
   - pugixml
-  - cpp-argparse
+  - cpp-argparse <3.1
   # Test dependencies
   - pytest
   - jupyter_kernel_test

--- a/environment-host.yml
+++ b/environment-host.yml
@@ -11,7 +11,7 @@ dependencies:
   - cppzmq
   - xtl=0.7
   - pugixml
-  - cpp-argparse <3.1
+  - cpp-argparse
   # Test dependencies
   - pytest
   - jupyter_kernel_test

--- a/src/xmagics/executable.cpp
+++ b/src/xmagics/executable.cpp
@@ -42,9 +42,8 @@
 
 namespace xcpp
 {
-    argparser executable::get_options()
+    static void get_options(argparser &argpars)
     {
-        argparser argpars("executable", XEUS_CLING_VERSION, argparse::default_arguments::none);
         argpars.add_description("write executable");
         argpars.add_argument("filename")
             .help("filename")
@@ -67,7 +66,6 @@ namespace xcpp
             .help("shows help message")
             .implicit_value(true)
             .nargs(0);
-        return argpars;
     }
 
     std::string executable::generate_fns(const std::string& cell,
@@ -265,7 +263,8 @@ namespace xcpp
 
     void executable::operator()(const std::string& line, const std::string& cell)
     {
-        auto argpars = get_options();
+        argparser argpars("executable", XEUS_CLING_VERSION, argparse::default_arguments::none);
+        get_options(argpars);
         argpars.parse(line);
 
         std::string ExeFile = argpars.get<std::string>("filename");

--- a/src/xmagics/executable.hpp
+++ b/src/xmagics/executable.hpp
@@ -25,7 +25,6 @@ namespace xcpp
     public:
 
         executable(cling::Interpreter& i) : m_interpreter(i) {}
-        argparser get_options();
         virtual void operator()(const std::string& line, const std::string& cell) override;
 
     private:

--- a/src/xmagics/execution.cpp
+++ b/src/xmagics/execution.cpp
@@ -33,9 +33,8 @@ namespace xcpp
         compilation_result = m_interpreter->process(init_timeit.c_str());
     }
 
-    argparser timeit::get_options()
+    void timeit::get_options(argparser &argpars)
     {
-        argparser argpars("timeit", XEUS_CLING_VERSION, argparse::default_arguments::none);
         argpars.add_description("Time execution of a C++ statement or expression");
         argpars.add_argument("-n", "--number")
             .help("execute the given statement n times in a loop. If this value is not given, a fitting value is chosen")
@@ -62,7 +61,6 @@ namespace xcpp
             .help("shows help message")
             .implicit_value(true)
             .nargs(0);
-        return argpars;
     }
 
     std::string timeit::inner(std::size_t number, const std::string& code) const
@@ -103,7 +101,8 @@ namespace xcpp
         // std::vector<std::string> results((std::istream_iterator<std::string>(iss)),
         //                          std::istream_iterator<std::string>());
 
-        auto argpars = get_options();
+        argparser argpars("timeit", XEUS_CLING_VERSION, argparse::default_arguments::none);
+        get_options(argpars);
         argpars.parse(line);
 
         // TODO find a way to use std::size_t

--- a/src/xmagics/execution.hpp
+++ b/src/xmagics/execution.hpp
@@ -44,7 +44,7 @@ namespace xcpp
 
         cling::Interpreter* m_interpreter;
 
-        argparser get_options();
+        void get_options(argparser &argpars);
         std::string inner(std::size_t number, const std::string& code) const;
         std::string _format_time(double timespan, std::size_t precision) const;
         void execute(std::string& line, std::string& cell);

--- a/src/xmagics/os.cpp
+++ b/src/xmagics/os.cpp
@@ -20,9 +20,8 @@
 
 namespace xcpp
 {
-    argparser writefile::get_options()
+    static void get_options(argparser &argpars)
     {
-        argparser argpars("file", XEUS_CLING_VERSION, argparse::default_arguments::none);
         argpars.add_description("write file");
         argpars.add_argument("-a", "--append")
             .help("append")
@@ -41,12 +40,12 @@ namespace xcpp
             .help("shows help message")
             .implicit_value(true)
             .nargs(0);
-        return argpars;
     }
 
     void writefile::operator()(const std::string& line, const std::string& cell)
     {
-        auto argpars = get_options();
+        argparser argpars("file", XEUS_CLING_VERSION, argparse::default_arguments::none);
+        get_options(argpars);
         argpars.parse(line);
 
         auto filename = argpars.get<std::string>("filename");

--- a/src/xmagics/os.hpp
+++ b/src/xmagics/os.hpp
@@ -21,7 +21,6 @@ namespace xcpp
     {
     public:
 
-        argparser get_options();
         virtual void operator()(const std::string& line, const std::string& cell) override;
 
     private:

